### DISCUSSION TITLE
for c/cpp source files, how to use keyboard shortcuts to compile and run

### DISCUSSION THREAD

**Discussion by ssh352** [LazyVim/LazyVim#4324] [Q&A]
In Rust,  rustacean.nvim allows me to use ctrl-option-r to run a single rs file

<img width="845" alt="Screenshot 2024-08-19 at 16 33 23" src="https://github.com/user-attachments/assets/4022787d-a7ff-4747-af30-72d6ba550af8">

In C/Cpp, could you please advise which plugin is in charge of c/cpp source file runnables? which plugins should I be looking at? thanks!

**Comment by rogeriofrsouza** [comment]
There's an extra for CLang: https://www.lazyvim.org/extras/lang/clangd
It uses `clangd_extensions`, check it docs if there's something you can setup https://github.com/p00f/clangd_extensions.nvim

**Reply by ssh352** [reply]
thanks, just checked clangd_extensions.nvim doesn't have this functionality. so How do c++/c people run single source file in Lazyvim?
